cmd/kops/integration_test.go

```diff
@@ -642,6 +643,15 @@ func TestPrivateFlannel(t *testing.T) {
 		runTestTerraformAWS(t)
 }
 
+// TestPrivateKindnet runs the test on a configuration with private topology, flannel networking
+func TestPrivateKindnet(t *testing.T) {
```

Comment:
how do I generate the files used by the test?

Reply:
Create a file at `tests/integration/update_cluster/privatekindnet/in-v1alpha2.yaml`

You can copy [this one](https://github.com/kubernetes/kops/blob/master/tests/integration/update_cluster/minimal-aws/in-v1alpha2.yaml), changing the cluster name to `privatekindnet.example.com`, replacing `spec.networking.cni` with `spec.networking.kindnet`, and setting whichever `KindnetNetworkingSpec` fields you want.

Then run `./hack/update-expected.sh` and it will generate the test outputs including the rendered addon manifest.

Reply:
👍